src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -95,73 +94,178 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+      if (teamInSection.has(request.team) && teamInSection.get(request.team) !== request.section) {
+        this.statusMessageService.showErrorMessage(
+            `Team "${request.team}" is detected in both Section "${request.section}"
+             and Section "${teamInSection.get(request.team)}".
+             Please use different team names in different sections`);
+        return;
+      }
+
+      teamInSection.set(request.team, request.section);
+    }
+
+    this.studentService.enrollStudents(this.courseid, studentsEnrollRequest).subscribe((resp: Students) => {
+      const enrolledStudents: Student[] = resp.students;
+      this.showEnrollResults = true;
+      this.statusMessage.pop(); // removes any existing error status message
+      this.statusMessageService.showSuccessMessage('Enrollment successful. Summary given below.');
+      this.enrollResultPanelList =
+          this.propagateEnrollResultPanelList(this.existingStudents, enrolledStudents,
+              studentsEnrollRequest.studentEnrollRequests);
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessage.pop(); // removes any existing error status message
+      this.statusMessageService.showErrorMessage(resp.error.message);
+    });
+    this.studentService.getStudentsFromCourse(this.courseid).subscribe((resp: Students) => {
+      this.existingStudents = resp.students;
+    });
+  }
+
+  private propagateEnrollResultPanelList(existingStudents: Student[], enrolledStudents: Student[],
+                                         enrollRequests: StudentEnrollRequest[]): EnrollResultPanel[] {
+    const updateStatus: { [key: string]: number } = {
+      new: 0,
+      modified: 1,
+      modifiedUnchanged: 2,
+      error: 3,
+      unmodified: 4,
     };
-    this.httpRequestService.put('/students', paramMap, this.enrollData)
-        .subscribe((resp: EnrollResultPanelList) => {
-          this.showEnrollResults = true;
-          this.statusMessage.pop(); // removes any existing error status message
-          this.statusMessageService.showSuccessMessage('Enrollment successful. Summary given below.');
-          this.enrollResultPanelList = resp.enrollResultPanelList;
-        }, (resp: ErrorMessageOutput) => {
-          this.statusMessage.pop(); // removes any existing error status message
-          this.statusMessage.push({
-            message: resp.error.message,
-            color: 'danger',
-          });
+    const panelClass: { [key: number]: string } = {
+      0: 'bg-primary',
+      1: 'bg-warning',
+      2: 'bg-info',
+      3: 'bg-danger',
+      4: 'bg-default',
+    };
+
+    const panels: EnrollResultPanel[] = [];
+    const studentLists: Student[][] = [];
+
+    for (const _ of Object.values(updateStatus)) {
+      studentLists.push([]);
+    }
+
+    // Identify students not in the enroll list.
+    for (const existingStudent of existingStudents) {
+      let isModified: boolean = false;
+      for (const enrolledStudent of enrolledStudents) {
```

Comment:
There is a function called find: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find